consensus/hotstuff/eventhandler/event_handler.go

```diff
@@ -243,6 +261,10 @@ func (e *EventHandler) OnPartialTcCreated(partialTC *hotstuff.PartialTcCreated)
 // be executed by the same goroutine that also calls the other business logic
 // methods, or concurrency safety has to be implemented externally.
 func (e *EventHandler) Start(ctx context.Context) error {
```

Comment:
Observation: we propose first (if we are primary) and _then_ start the pacemaker for the respective round 
https://github.com/onflow/flow-go/blob/d74cd32abc0770f437da1b22c9e0e98ffeda662e/consensus/hotstuff/eventhandler/event_handler.go#L272-L276
In my mind, proposing is part of the round. Therefore, I would PaceMaker should be started _before_ we try to propose.